src/pages/EditQuestion.jsx

```diff
@@ -139,8 +156,8 @@ export default function EditQuestion({ question, onUpdated }) {
       <Box my={2}>
         <NewDownloadButton
           displayText="Download JSON"
-          getData={() => API.getQuestionData(question.id, user.id_token)}
-          fileName="question_data.json"
+          getData={() => API.getQuestionData(question.id, token)}
+          fileName={questionFileName()}
```

Comment:
If question name doesn't exist, this crashes the page. We also shouldn't be passing this as a called function. This will call this function on every render. I would pass it as a function (drop the parentheses) and just call the function when we actually want to download the message.